src/scripts/main.js

```diff
@@ -1 +1,84 @@
 'use strict';
+
+const button = document.querySelector('body');
+
+button.addEventListener('contextmenu', (e) => {
+  e.preventDefault();
+});
+
+const firstPromise = new Promise((resolve, reject) => {
+  setTimeout(reject, 3000);
+
+  button.addEventListener('click', resolve);
+});
+
+const secondPromise = new Promise((resolve) => {
+  button.addEventListener('mouseup', (e) => {
+    switch (e.button) {
+      case 0:
+        resolve();
+        break;
+      case 2:
+        e.preventDefault();
+        resolve();
```

Comment:
The `e.preventDefault()` call on the right-click event might not be necessary unless you want to prevent the default context menu from appearing. Ensure this behavior is intended.